staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd2.go

```diff
@@ -17,33 +17,56 @@ limitations under the License.
 package factory
 
 import (
+	"context"
 	"net"
 	"net/http"
 	"time"
 
 	etcd2client "github.com/coreos/etcd/client"
 	"github.com/coreos/etcd/pkg/transport"
+	"github.com/golang/glog"
 
 	utilnet "k8s.io/apimachinery/pkg/util/net"
+	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/storage"
 	"k8s.io/apiserver/pkg/storage/etcd"
 	"k8s.io/apiserver/pkg/storage/storagebackend"
 )
 
 func newETCD2Storage(c storagebackend.Config) (storage.Interface, DestroyFunc, error) {
-	tr, err := newTransportForETCD2(c.CertFile, c.KeyFile, c.CAFile)
+	tr, err := NewTransportForETCD2(c.CertFile, c.KeyFile, c.CAFile)
```

Comment:
don't export these... if you need an endpoints getter, define the interface and implement it in `storagebackend/factory/factory.go`

Reply:
ok

Reply:
@liggitt fixed, ptal